tools/alara_output_processing/alara_output_plotting.py

```diff
@@ -209,6 +236,209 @@ def construct_legend(ax, data_comp=False):
+        times (list of floats): Chronological list of cooling times.
+        nrows (int): Number of rows in the grid of pie charts.
+        ncols (int): Number of columns in the grid of pie charts.
+        time_unit (str): Units for cooling times.
+        color_map (dict): Pre-constructed color map for each nuclide to match
+            table rows with pie chart wedges.
+
+    Returns:
+        gs (matplotlib.gridspec.GridSpec): Matplotlib Gridspec object to
+            organize the combined set of subplots containing the resultant
+            table as well as the individual pie charts, to be added
+            subsequent to this function's calling.
+    '''
+
+    # Create table gridspec
+    left = 0.05
+    top = 0.95
+    gs = fig.add_gridspec(
+        nrows + 1, ncols,
+        height_ratios=([1.0] + [1]*nrows),
+        hspace=0.4,
+        wspace=0.3,
+        left=left, right=(1-left), top=top, bottom=(1-top)
+    )
+
+    # Build and populate table
+    table_ax = fig.add_subplot(gs[0,:])
+    table_ax.axis('off')
+    table_data = populate_table_rows(agg, all_nucs, times)
+
+    col_labels = [f'{t} {time_unit}' for t in times]
+    row_labels = [reformat_isotope(nuc) for nuc in all_nucs]
+
+    table_data_float = np.array([
+        [float(cell.strip('%')) for cell in row] for row in table_data
+    ])
+    row_means = table_data_float.mean(axis=1)
+    sorted_indices = np.argsort(-row_means)
+    table_data = sort_by_index(table_data, sorted_indices)
+    row_labels = sort_by_index(row_labels, sorted_indices)
+
+    table = table_ax.table(
+        cellText=table_data,
+        rowLabels=row_labels,
+        colLabels=col_labels,
+        loc='center',
+        cellLoc='center'
+    )
+    table.auto_set_font_size(False)
+    table.set_fontsize(11)
+    table.scale(1.0, 2.0)
+
+    # Format table for readability
+    for i in range(len(row_labels)):
+        table[(i+1, -1)].get_text().set_fontweight('bold')
+    for j in range(len(col_labels)):
+        table[(0, j)].get_text().set_fontweight('bold')
+
+    for i, nuc in enumerate([all_nucs[idx] for idx in sorted_indices]):
+        rgba = color_map[nuc]
+        
+        # Color in index
+        table[(i+1, -1)].set_facecolor(rgba)
+        
+        # Color all other columns
+        for j in range(len(times)):
+            table[(i+1, j)].set_facecolor(rgba)
+
+    return gs
+
+def pie_labels_by_threshold(time_slice, threshold):
+    '''
+    Filter pie chart labels to group small contributions with the "Other"
+        label for readability.
+    
+    Arguments:
+        
+    '''
+
+    values = time_slice['value'].to_numpy()
+    total = time_slice['value'].sum()
+    fractions = values / total if total > 0 else np.zeros_like(values)
+
+    labels = []
+    if np.any(fractions == 1.0):
+        for nuc, frac in zip(time_slice['nuclide'], fractions):
+            labels.append(reformat_isotope(nuc) if frac == 1.0 else '')
+        return labels
```

Comment:
Isn't this edge case already covered by the case below, as long as you handle `Other` more carefully? Maybe the condition is:
```
if frac >= threshold or (nuc == 'Other' && frac > 0):
```